chasm/engine.go

```diff
@@ -1,10 +1,12 @@
+//go:generate mockgen -package $GOPACKAGE -source $GOFILE -destination engine_mock.go
+
 package chasm
 
 import (
 	"context"
 )
 
-type engine interface {
+type Engine interface {
 	newEntity(
```

Comment:
I think those methods needs to be exported as well or the ChasmEngine struct in history package won't implement this interface.